src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+        testInvalidLogin(INVALID_LOGIN, "shorter than 6 symbols");
+    }
+
+    @Test
+    void register_InvalidPasswordTest_NotOk() {
+        testInvalidPassword(null, "null");
+        testInvalidPassword("", "empty String");
+        testInvalidPassword(INVALID_PASSWORD, "shorter than 6 symbols");
+    }
+
+    @Test
+    void register_InvalidAgeTest_NotOk() {
+        testInvalidAge(null,"null");
+        testInvalidAge(MIN_VALID_AGE - 1,"< than 0");
+        testInvalidAge(INVALID_AGE_ZERO,"0");
+        testInvalidAge(INVALID_AGE_BELOW_ZERO,"< 0");
+    }
+
+    @Test
+    void register_ValidAgeTest_Ok() {
+        int testNumber = 0;
+
+        Integer minAge = MIN_VALID_AGE;
+        testValidAge(minAge, ++testNumber);
+        testValidAge(minAge + 1, ++testNumber);
+        testValidAge(minAge + 10, ++testNumber);
+    }
+
+    private void testValidAge(Integer age, int testNumber) {
+        User expected = createValidUser(testNumber);
+
+        expected.setAge(age);
+        User actual = registrationServiceImpl.register(expected);
+        assertEquals(expected, actual,
+                "User adding with walid age doesnt work; Age = " + age);
+    }
+
+    private void testInvalidLogin(String login, String sortOffInvalidLogin) {
+        User userWithInvalidLogin = createValidUser();
+        String message = "Login " + sortOffInvalidLogin
+                + " should throw UserNotValidException; Login is " + login;
+
+        userWithInvalidLogin.setLogin(login);
+        testInvalidUser(userWithInvalidLogin, message);
+    }
+
+    private void testInvalidPassword(String password, String sortOffInvalidPassword) {
+        User userWithInvalidPassword = createValidUser();
+        String message = "Password " + sortOffInvalidPassword
+                + " should throw UserNotValidException; Password is " + password;
+
+        userWithInvalidPassword.setPassword(password);
+        testInvalidUser(userWithInvalidPassword, message);
+    }
+
+    private void testInvalidAge(Integer age, String sortOfInvalidAge) {
+        User userWithInvalidAge = createValidUser();
+        String message = "Age " + sortOfInvalidAge
+                + " should throw UserNotValidException; Age = " + age;
+
+        userWithInvalidAge.setAge(age);
+        testInvalidUser(userWithInvalidAge, message);
+    }
+
+    private void testInvalidUser(User invalidUser, String message) {
+        assertThrows(UserNotValidException.class, () ->
+                registrationServiceImpl.register(invalidUser), message);
+    }
+
+    private User createValidUser(int indexOfUser) {
+        User user = createValidUser();
+        user.setLogin(user.getLogin() + indexOfUser);
+        return user;
+    }
+
+    private User createValidUser() {
+        User user = new User();
+        user.setLogin(SHORTEST_VALID_LOGIN);
+        user.setPassword(SHORTEST_VALID_PASSWORD);
+        user.setAge(MIN_VALID_AGE);
+        return user;
+    }
+}
```

Comment:
You don't need all these private methods, it will be enough to use only assert method